### PR TITLE
Answer:1 - angular content projection

### PR DESCRIPTION
**Pull request by alcaidio**
## Checklist for challenge submission

- [x] Start your PR message with Answer:${challenge_number}

## Some notes about the code 

- refactor state services with signal and reduce duplication with a generic service.
- removal of unnecessary folders
- use of new syntax (input, output, contentchild, controlflow) angular 17 
- use of ng-content and ngtemplateoutlet with typed context using a directive, inspired by [the article](https://medium.com/@thomas.laforge/ngtemplateoutlet-type-checking-5d2dcb07a2c6
) of @tomalaforge.
- **one problem**: can't use *appItem syntax shortcut (see teacher card)

### CODE REVIEW

File: apps/angular/projection/src/app/ui/card.component.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  imports: [NgTemplateOutlet, ItemDirective],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  host: {
+    class: 'flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4',
+  },
+})
+export class CardComponent<T extends { id: number }> {
+  list = input<T[] | null>(null);
+  addItem = output<void>();
+  itemTemplateRef = contentChild.required(ItemDirective, { read: TemplateRef });
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/card.component.ts:64]: 👍

File: apps/angular/projection/src/app/ui/card.component.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    ctx: unknown,
+  ): ctx is ItemContext<TContext> {
+    return true;
+  }
+}
+
+@Component({
+  selector: 'app-card',
+  template: `
+    <ng-content select="img"></ng-content>
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/card.component.ts:36]: 👍
you can also use the short synthax <ng-conent />

File: apps/angular/projection/src/app/ui/list-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  ChangeDetectionStrategy,
+  Component,
+  input,
+  output,
+} from '@angular/core';
+
+@Component({
+  selector: 'app-list-item',
+  template: `
+    {{ label() }}
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/list-item.component.ts:11]: use ng-content here

File: apps/angular/projection/src/app/ui/list-item.component.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    </button>
+  `,
+  standalone: true,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  host: {
+    class: 'border-grey-300 flex justify-between border px-2 py-1',
+  },
+})
+export class ListItemComponent {
+  id = input.required<number>();
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/list-item.component.ts:23]: you don't need to pass the id, the output is just an event emitter

File: apps/angular/projection/src/app/component/student-card.component.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+export class StudentCardComponent {
+  private http = inject(FakeHttpService);
+  protected store = inject(StudentStore);
+  randStudent = () => randStudent();
+
+  constructor() {
+    this.http.fetchStudents$
+      .pipe(
+        tap((s) => this.store.addAll(s)),
+        takeUntilDestroyed(),
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/component/student-card.component.ts:46]: 👍

**Comment by tomalaforge** [comment]
very good and very complete solution

File: apps/angular/projection/src/app/ui/card.component.ts (hunk 2 above)
Comment (alcaidio) [reply apps/angular/projection/src/app/ui/card.component.ts:36]: What is the short syntax for   `<ng-content select="img"></ng-content>
` ?

**Pull request closed without merging**